drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1729,12 +1660,10 @@ static int eth_initialize(const struct device *dev)
 	return retval;
 }
 
-#if DT_INST_NODE_HAS_PROP(0, mac_eeprom)
-static void get_mac_addr_from_i2c_eeprom(uint8_t mac_addr[6])
+static void get_mac_addr_from_i2c_eeprom(uint8_t mac_addr[6], const struct i2c_dt_spec i2c)
```

Comment:
is this part documented in the datasheet or other documentation of the soc manufacturer?
If not it should be removed, as it's up to the user to to set the mac later f.e. via the set_config api. 

Also having ETH_SAM_GMAC_MAC_I2C_INT_ADDRESS and ETH_SAM_GMAC_MAC_I2C_INT_ADDRESS_SIZE as a Kconfig is not a good idea, as it can this way only be set for all instances at one, so for every instance another eeprom has to be used.

Reply:
if it's not documented in a datasheet or a document for the soc, it should be removed, we had a similar case here
https://github.com/zephyrproject-rtos/zephyr/pull/87953#discussion_r2022779300